### PR TITLE
Update the Knative installation guide

### PR DESCRIPTION
**Pull request by zhenghuiwang**
**What this PR does / why we need it**:
Add instructions on how to install Knative Serving via kubeflow/manifests. The original guide on installing knative v0.9 doesn't work with Kubeflow v0.6.2.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/359)

### CODE REVIEW

File: docs/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -52,12 +52,17 @@ You must install these tools:
...
 
-- **Note**: KFServing currently only requires `Knative Serving` for auto-scaling, canary rollout,
- `Istio` for traffic routing and ingress. You can follow instructions on
- [Custom Install](https://knative.dev/docs/install/knative-custom-install) to install `Istio` and `Knative Serving`,
- observability plug-ins are good to have for monitoring. If you already have `Istio` (e.g. from a kubeflow install) then simply skip the `Istio` steps.
+* You can follow the instructions on [Set up a kubernetes cluster and install Knative Serving](https://knative.dev/docs/install/) or
+[Custom Install](https://knative.dev/docs/install/knative-custom-install) to install `Istio` and `Knative Serving`. Observability plug-ins are good to have for monitoring.
+
+* If you already have `Istio` (e.g. from a Kubeflow install) then simply skip the `Istio` steps. For Kubeflow install, you can install `Knative Serving` via
+the following commands after downloading repository [kubeflow/manifests](https://github.com/kubeflow/manifests).
```
Comment (johnugeorge) [inline docs/DEVELOPER_GUIDE.md]: May be specify that version of knative in manifests is 0.8

Comment (zhenghuiwang) [reply docs/DEVELOPER_GUIDE.md]: Done

Comment (yuzisun) [reply docs/DEVELOPER_GUIDE.md]: Any reason why native 0.9 does not work with kubeflow 0.6.2?

Comment (zhenghuiwang) [reply docs/DEVELOPER_GUIDE.md]: The `autoscaler` pod of Knative 0.9 is always in CrashBackOff loop after being installed on Kubeflow v0.6.2.

I don't know the reason as no logs is shown for the pod. but my guess is there is some Istio setting of KF doesn't work with knative 0.9 out of the box. That's why I start look into manifests repo to see if there some knative config that works with current KF.

**Comment by ellis-bigelow** [comment]
Awesome thanks for this. Can you also add a link on the frontpage readme? Perhaps in the installation instructions there? Users get hung up there a lot.

File: docs/DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -52,12 +52,17 @@ You must install these tools:
...
- `Istio` for traffic routing and ingress. You can follow instructions on
- [Custom Install](https://knative.dev/docs/install/knative-custom-install) to install `Istio` and `Knative Serving`,
- observability plug-ins are good to have for monitoring. If you already have `Istio` (e.g. from a kubeflow install) then simply skip the `Istio` steps.
+* You can follow the instructions on [Set up a kubernetes cluster and install Knative Serving](https://knative.dev/docs/install/) or
+[Custom Install](https://knative.dev/docs/install/knative-custom-install) to install `Istio` and `Knative Serving`. Observability plug-ins are good to have for monitoring.
+
+* If you already have `Istio` (e.g. from a Kubeflow install) then simply skip the `Istio` steps. For Kubeflow install, you can install `Knative Serving` v0.8 via
+the following commands after downloading repository [kubeflow/manifests](https://github.com/kubeflow/manifests).
+
+  ``` kubeflow/manifests/knative/knative-serving-crd/base$ kustomize build . | kubectl apply -f -```
```
Comment (shotarok) [inline docs/DEVELOPER_GUIDE.md]: [nits] `s/knative-serving-crd/knative-serving-crds/`

Comment (zhenghuiwang) [reply docs/DEVELOPER_GUIDE.md]: Done. thanks

**Comment by zhenghuiwang** [comment]
> Awesome thanks for this. Can you also add a link on the frontpage readme? Perhaps in the installation instructions there? Users get hung up there a lot.

Done.

File: docs/DEVELOPER_GUIDE.md (hunk 1 above)
Comment (yuzisun) [reply docs/DEVELOPER_GUIDE.md]: @zhenghuiwang you should be able to get the logs with `--previous` flag, I am curious about the error. Also which version of Istio kubeflow is using ?

**Pull request merged**